server/tools/peertube-auth.ts

```diff
@@ -0,0 +1,122 @@
+  netrc.save()
+}
+
+const isURLaPeerTubeInstance = (url) => {
+  return isHostValid(url)
+}
+
+program
+  .name('auth')
+  .usage('[command] [options]')
+
+program
+  .command('add')
+  .description('remember your accounts on remote instances for easier use')
+  .option('-u, --url <url>', 'Server url')
+  .option('-U, --username <username>', 'Username')
+  .option('-p, --password <token>', 'Password')
+  .option('--default', 'add the entry as the new default')
+  .action(options => {
+    prompt.override = options
+    prompt.start()
+    prompt.get({
+      properties: {
+        url: {
+          description: 'instance url',
+          conform: (value) => isURLaPeerTubeInstance(value),
+          required: true
+        },
+        username: {
+          pattern: /^[a-zA-Z\s\-]+$/,
+          message: 'Name must be only letters, spaces, or dashes',
+          required: true
+        },
+        password: {
+          hidden: true,
+          required: true
+        }
+      }
+    }, (_, result) => {
+      settings.remotes.splice(settings.remotes.indexOf(result.url))
+      settings.remotes.push(result.url)
+      config.write(settings, function (err) {
+        if (err) console.error(err)
+      })
+      setPassword(result.url, result.username, result.password)
+    })
+  })
+
+program
+  .command('del <url>')
+  .description('unregisters a remote instance')
+  .action((url) => {
+    settings.remotes.splice(settings.remotes.indexOf(url))
+    config.write(settings, function (err) {
+      if (err) console.error(err)
+    })
+
+    netrc.loadSync()
+    delete netrc.machines[url]
+    netrc.save()
+  })
+
+program
+  .command('list')
+  .description('lists registered remote instances')
+  .action(() => {
+    config.read((_, data) => {
+      const table = new Table({
+        head: ['instance', 'login'],
+        colWidths: [30, 30]
+      })
+      netrc.loadSync()
+      data.remotes.forEach(element => {
+        table.push([ element, netrc.machines[element].login ])
+      })
+
+      console.log(table.toString())
+    })
+  })
+
+program
+  .command('set-default <url>')
+  .description('set an existing entry as default')
+  .action((url) => {
+    config.read((_, data) => {
+      if (data.remotes.indexOf(url) !== -1) {
+        data.default = data.remotes.indexOf(url)
+        config.write(data, function (err) {
+          if (err) console.error(err)
+        })
+      } else {
+        console.log('<url> is not a registered instance.')
+        process.exit(-1)
+      }
+    })
+  })
+
+program.on('--help', function () {
+  console.log('  Examples:')
+  console.log()
+  console.log('    $ peertube add -u peertube.cpy.re -U "PEERTUBE_USER" --password "PEERTUBE_PASSWORD"')
+  console.log('    $ peertube add -u peertube.cpy.re -U root')
+  console.log('    $ peertube list')
+  console.log('    $ peertube del peertube.cpy.re')
+  console.log()
+})
+
+if (!process.argv.slice(2).length) {
```

Comment:
Why do you use slice instead of `process.argv..length !== 2`?

Reply:
Just because it was the example provided in commanderjs's doc and it JustWorked™ :wink:

Reply:
Please use `process.argv.length !== 2`, I don't get the point of the commanderjs doc version